lambda-buffers-compiler/app/LambdaBuffers/Compiler/Cli/Compile.hs

```diff
@@ -30,8 +41,8 @@ compile opts = do
       ProtoError pe -> print $ "Encountered a proto error " <> show pe
     Right compIn' -> do
       print @String "Successfully processed the CompilerInput"
-      writeCompilerOutput (opts ^. output) (toProto compIn')
-
+      let result = either (RCompilerFailure . KCErr) (RCompilerOutput . CompilerOutput . fmap kindConvert . getAllContext) $ check compIn'
```

Comment:
This is hard to read. What I want the CLI code to look like is to call `KindCheck.check compIn`, all the context building should happen in that function in the KindCheck module.